instrumentation/internal/internal-reflection/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/internal/reflection/ReflectionHelper.java

```diff
@@ -36,6 +38,7 @@ private ReflectionHelper() {}
       return methods;
     } else if (containingClass.isInterface()
```

Comment:
If filtering of the interfaces works then this branch shouldn't evaluate to true. Similarly I suspect that  https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/internal/internal-proxy/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/internal/proxy/ProxyHelper.java might not be necessary any more.

Reply:
nice, I'll open an issue to try this 👍